src/particles/wakefields/WakeConvolution.cpp

```diff
@@ -0,0 +1,170 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Alex Bojanich, Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "WakeConvolution.H"
+#include "particles/ImpactXParticleContainer.H"
+
+#ifdef ImpactX_USE_FFT
+#include <ablastr/math/fft/AnyFFT.H>
+#endif
+
+#include <AMReX_REAL.H>
+
+#include <algorithm>
+#include <cmath>
+#ifndef ImpactX_USE_FFT
+#include <stdexcept>
+#endif
+
+namespace impactx::particles::wakefields
+{
+    amrex::Real alpha (amrex::Real s)
+    {
+        using namespace amrex::literals;
+
+        return 1_rt - impactx::particles::wakefields::alpha_1 * std::sqrt(s) - (1_rt - 2_rt * impactx::particles::wakefields::alpha_1) * s;
+    }
+
+    amrex::Real w_t_rf (
+        amrex::Real s,
+        amrex::Real a,
+        amrex::Real g,
+        amrex::Real L
+    )
+    {
+        using namespace amrex::literals;
+
+        amrex::Real const s0 = (0.169_rt * std::pow(a, 1.79_rt) * std::pow(g, 0.38_rt)) / std::pow(L, 1.17_rt);
+        amrex::Real const term = std::sqrt(std::abs(s) / s0) * std::exp(-std::sqrt(std::abs(s) / s0));
+        return (4 * impactx::particles::wakefields::Z0 * ablastr::constant::SI::c * s0 * unit_step(s)) / (amrex::Real(M_PI) * std::pow(a, 4)) * term;
+    }
+
+    amrex::Real w_l_rf (
+        amrex::Real s,
+        amrex::Real a,
+        amrex::Real g,
+        amrex::Real L
+    )
+    {
+        using namespace amrex::literals;
+
+        amrex::Real const s00 = g * std::pow((a / (alpha(g / L) * L)), 2) / 8.0_rt;
+        return (impactx::particles::wakefields::Z0 * ablastr::constant::SI::c * unit_step(s) * std::exp(-std::sqrt(std::abs(s) / s00))) / (amrex::Real(M_PI) * std::pow(a, 2));
+    }
+
+    void convolve_fft (
+        amrex::Gpu::DeviceVector<amrex::Real> const & beam_profile_slope,
+        amrex::Gpu::DeviceVector<amrex::Real> const & wake_func,
+        amrex::Real delta_t,
+        amrex::Gpu::DeviceVector<amrex::Real> & result,
+        int padding_factor
+    )
+    {
+    #ifdef ImpactX_USE_FFT
+        int const beam_profile_slope_size = beam_profile_slope.size();
+        int const wake_func_size = wake_func.size();
+
+        // Length of convolution result
+        int const original_n = beam_profile_slope_size + wake_func_size - 1;  // Output size is n = 2N - 1, where N = size of signals 1,2
```

Comment:
```suggestion
        int const original_n = beam_profile_slope_size + wake_func_size;  // Output size is n = 2*num_bins - 1, where N = size of signals 1,2 and slope has one element less
```

Reply:
No, needs to be complex size = N/2 + 1 now :
https://www.fftw.org/fftw3_doc/One_002dDimensional-DFTs-of-Real-Data.html

Reply:
Fixed.